### PR TITLE
Dev/fix warnigs

### PR DESCRIPTION
**Pull request by Diaskhan**
fix warnigs
fix misc issues
fix beahviour of One instance(that suppress errors in main thread)
Added option AllowOnlyOneInstance

https://github.com/anovik/SmartCommander/issues/89

### CODE REVIEW

**Comment by Diaskhan** [comment]
Its better add a option to settings, allowing one or more instances, instead of forcing use One instance

File: src/SmartCommander/Models/OptionsModel.cs [hunk 1]
```diff
@@ -48,11 +48,12 @@ static OptionsModel()
 
         public bool IsMaximized { get; set; }
 
-        public string LeftPanePath { get; set; } = "";
+        public string LeftPanePath { get; set; } = string.Empty;
```
Comment (anovik) [inline src/SmartCommander/Models/OptionsModel.cs:52]: I don't think that string.Empty is more readable. Would prefer to leave "" in all cases.

File: src/SmartCommander/Models/OptionsModel.cs [hunk 2]
```diff
@@ -48,11 +48,12 @@ static OptionsModel()
...
 
-        public string RightPanePath { get; set; } = "";
+        public string RightPanePath { get; set; } = string.Empty;
 
         public bool IsDarkThemeEnabled { get; set; }
+        public bool AllowOnlyOneInstance { get; set; }
```
Comment (anovik) [inline src/SmartCommander/Models/OptionsModel.cs:56]: Would prefer one instance to be the behavior by default.

File: src/SmartCommander/ViewModels/OptionsViewModel.cs [hunk 3]
```diff
@@ -51,10 +51,11 @@ public OptionsViewModel()
             ConfirmationWhenDeleteNonEmpty = Model.ConfirmationWhenDeleteNonEmpty;
             SaveWindowPositionSize = Model.SaveWindowPositionSize;
             IsDarkThemeEnabled = Model.IsDarkThemeEnabled;
+            AllowOnlyOneInstance=Model.AllowOnlyOneInstance;
```
Comment (anovik) [inline src/SmartCommander/ViewModels/OptionsViewModel.cs:54]: whitespaces are missing before and after =

**Review by anovik — COMMENTED** [review]
Please have a look at the comments. And again, I would prefer separate PRs for warnings and for allow only one instance change. Regarding the latest, default behavior should be one instance only.

File: src/SmartCommander/Models/OptionsModel.cs (hunk 1 above)
Comment (Diaskhan) [reply src/SmartCommander/Models/OptionsModel.cs]: fixed

File: src/SmartCommander/ViewModels/OptionsViewModel.cs (hunk 3 above)
Comment (Diaskhan) [reply src/SmartCommander/ViewModels/OptionsViewModel.cs:54]: fixed

File: src/SmartCommander/Models/OptionsModel.cs (hunk 2 above)
Comment (Diaskhan) [reply src/SmartCommander/Models/OptionsModel.cs:56]: fixed

**Review by anovik — APPROVED** [review]

**Pull request merged**